Neolution.DotNet.Console/DotNetConsole.cs

```diff
@@ -1,73 +1,299 @@
+                .AddJsonFile($"appsettings.{environment.EnvironmentName}.json", optional: true, reloadOnChange: true);
+
+            if (environment.IsDevelopment())
+            {
+                configurationBuilder.AddUserSecrets(assembly, optional: true, reloadOnChange: true);
+            }
+
+            configurationBuilder.AddEnvironmentVariables();
+
+            return configurationBuilder.Build();
+        }
+
+        /// <summary>
+        /// Adjusts the default builder logging providers.
+        /// </summary>
+        /// <param name="logging">The logging.</param>
+        private static void AdjustDefaultBuilderLoggingProviders(ILoggingBuilder logging)
+        {
+            // Remove the default logging providers
+            logging.ClearProviders();
+
+            // Re-add other logging providers that are assigned in Host.CreateDefaultBuilder
+            logging.AddDebug();
+            logging.AddEventSourceLogger();
+
+            if (OperatingSystem.IsWindows())
+            {
+                // Add the EventLogLoggerProvider on windows machines
+                logging.AddEventLog();
+            }
+        }
+
+        /// <summary>
+        /// Adds the command line configuration.
+        /// </summary>
+        /// <param name="configBuilder">The configuration builder.</param>
+        /// <param name="args">The arguments.</param>
+        private static void AddCommandLineConfig(IConfigurationBuilder configBuilder, string[]? args)
+        {
+            if (args is { Length: > 0 })
+            {
+                configBuilder.AddCommandLine(args);
+            }
+        }
+
+        /// <summary>
+        /// Runs the command with options asynchronously.
+        /// </summary>
+        /// <param name="options">The options.</param>
+        /// <returns>The <see cref="Task"/>.</returns>
+        private async Task RunWithOptionsAsync(object options)
+        {
+            // To support scoped services, create a scope for each command call/run.
+            var scopeFactory = this.Services.GetRequiredService<IServiceScopeFactory>();
+            await using var scope = scopeFactory.CreateAsyncScope();
+
+            // Determine the type of the options object to find the corresponding command type
+            var dataType = new[] { options.GetType() };
+            var genericBase = typeof(IDotNetConsoleCommand<>);
+            var commandType = genericBase.MakeGenericType(dataType);
+            var commandEntryPointMethodInfo = commandType.GetMethod(nameof(IDotNetConsoleCommand<object>.RunAsync));
+
+            // Resolve the command from the service provider by the determined type and invoke the entry point method (RunAsync)
+            var command = scope.ServiceProvider.GetRequiredService(commandType);
+            var result = commandEntryPointMethodInfo?.Invoke(command, new[] { options }) as Task;
+            if (result is null)
+            {
+                await Task.CompletedTask;
```

Comment:
is this "await Task.CompletedTask" really necessary? what is the purpose?

Reply:
Nope, I've overseen that one. Fixed.